pkg/ctl/delete/nodegroup.go

```diff
@@ -56,31 +56,36 @@ func deleteNodeGroupCmd(g *cmdutils.Grouping) *cobra.Command {
 	return cmd
 }
 
-func doDeleteNodeGroup(p *api.ProviderConfig, cfg *api.ClusterConfig, ng *api.NodeGroup, nameArg string) error {
-	ctl := eks.New(p, cfg)
+func doDeleteNodeGroup(p *api.ProviderConfig, cfg *api.ClusterConfig, ng *api.NodeGroup, nameArg string, cmd *cobra.Command) error {
+	cfgLoader := cmdutils.NewClusterConfigLoader(p, cfg, clusterConfigFile, cmd)
 
-	if err := api.Register(); err != nil {
-		return err
-	}
+	cfgLoader.ValidateWithoutConfigFile = func() error {
+		if cfg.Metadata.Name == "" {
+			return cmdutils.ErrMustBeSet("--cluster")
+		}
+		if ng.Name != "" && nameArg != "" {
+			return cmdutils.ErrNameFlagAndArg(ng.Name, nameArg)
+		}
 
-	if err := ctl.CheckAuth(); err != nil {
-		return err
-	}
+		if nameArg != "" {
+			ng.Name = nameArg
+		}
 
-	if cfg.Metadata.Name == "" {
-		return errors.New("--cluster must be set")
-	}
+		if ng.Name == "" {
+			return cmdutils.ErrMustBeSet("--name")
+		}
 
-	if ng.Name != "" && nameArg != "" {
-		return cmdutils.ErrNameFlagAndArg(ng.Name, nameArg)
+		return nil
 	}
 
-	if nameArg != "" {
-		ng.Name = nameArg
+	if err := cfgLoader.Load(); err != nil {
```

Comment:
TODO: I think this one actually needs much more work, will leave it out for another PR